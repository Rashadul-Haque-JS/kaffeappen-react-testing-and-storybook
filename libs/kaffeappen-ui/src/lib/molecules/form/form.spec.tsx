import { render } from '@testing-library/react';
import FormComponent from './form';
import ButtonComponent from '../../atoms/button/button';
import InputComponent from '../../atoms/input/input';

describe('Form', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <FormComponent>
        <InputComponent placeholder="användernamn" type={'text'} />
        <InputComponent placeholder="lösenord" type={'password'} />
        <ButtonComponent children="Logga In" variant="primary" marginTop={40} />
      </FormComponent>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot successfully', () => {
    const { baseElement } = render(
      <FormComponent>
        <InputComponent placeholder="användernamn" type={'text'} />
        <InputComponent placeholder="lösenord" type={'password'} />
        <ButtonComponent children="Logga In" variant="primary" marginTop={40} />
      </FormComponent>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
