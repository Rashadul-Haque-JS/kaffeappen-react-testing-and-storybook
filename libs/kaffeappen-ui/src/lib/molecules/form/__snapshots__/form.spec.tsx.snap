// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot successfully 1`] = `
<body>
  <div>
    <form
      class="form__StyledForm-qsc0mr-0 eqcWSg"
    >
      <input
        class="input__StyledInput-sc-1rotb0l-0 HWiIs"
        placeholder="användernamn"
        required=""
        type="text"
      />
      <input
        class="input__StyledInput-sc-1rotb0l-0 HWiIs"
        placeholder="lösenord"
        required=""
        type="password"
      />
      <button
        class="button__StyledButton-sc-1l6wl6y-0 jakkFT"
      >
        Logga In
      </button>
    </form>
  </div>
</body>
`;
